### PR TITLE
Update linter

### PR DESCRIPTION
**Pull request by anyasabo**
I ran into a bug when troubleshooting the linter failure in https://github.com/elastic/cloud-on-k8s/pull/2724 that is resolved in newer golangci-lint versions (a lot of discussion in https://github.com/golangci/golangci-lint/issues/827).

Fixes a bunch of linter warnings:
- Corrects duplicate imports and ensures receiver names are uniform.
- I also added an exception for cyclomatic complexity in one of our init functions because I think it's okay in that particular area.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ linters:
     - scopelint
     - whitespace # added in 1.19
     - wsl # added in 1.20
+    - gomnd # added in 1.23
```
Comment (charith-elastic) [inline .golangci.yml]: Nit: it would be nice to maintain the sorted order in this list

File: .golangci.yml [hunk 2]
```diff
@@ -98,6 +99,5 @@ issues:
       text: 'G101: Potential hardcoded credentials'
     - path: pkg/controller/elasticsearch/driver/version\.go
       text: 'Consider preallocating `vs`'
-    - linters:
-        - stylecheck
-      text: 'ST1016: methods on the same type should have the same receiver name.*"in".*'
+    - path: cmd/manager/main\.go
```
Comment (charith-elastic) [inline .golangci.yml]: Again, this list was roughly sorted by path. No big deal though.

Comment (anyasabo) [reply .golangci.yml]: Ah I didn't realize for either of them

**Comment by anyasabo** [comment]
jenkins test this please

**Comment by anyasabo** [comment]
I actually changed this to the next to latest version of golangci-lint because it is OOMing jenkins. Already open issue here:
https://github.com/golangci/golangci-lint/issues/994

And example test run failure of previous version of this PR with 1.24.0:
https://devops-ci.elastic.co/job/cloud-on-k8s-pr/3002/consoleFull

**Comment by anyasabo** [comment]
jenkins test this please

**Comment by anyasabo** [comment]
jenkins test this please

**Pull request merged**